miner/worker.go

```diff
@@ -287,6 +294,31 @@ func (miner *Miner) commitTransaction(env *environment, tx *types.Transaction) e
 	if tx.Type() == types.BlobTxType {
 		return miner.commitBlobTransaction(env, tx)
 	}
+
+	// If a conditional is set, check prior to applying
```

Comment:
Have you considered putting the rate limiter in the RPC layer? Just curious if you have thought about it at all. It seems like it would be better UX for bundlers to learn that their transactions is not gonna make it earlier rather than it being silently dropped in the miner

Reply:
It also saves some resources as the transaction would be rejected directly higher up in the stack. Now that I think about it more, i feel strongly that its a better idea to put this in the RPC layer

Reply:
It makes sense if the intention is not to arbitrarily abort transactions this far down, but does it expand the scope of the PR? Can it be done in a later PR rather than block this one?

Reply:
The check is a last ditch effort for transactions that race and invalidate each other in the same block. We have metrics here to detect how often this happens.

But this check is absolutely done in the RPC layer, `sequencerapi`. And also there's support for sentry nodes to do this check as well since these checks also occur there prior to forwarding to the sequencer url.

In order to completely do it in the RPC layer, there would have to be a way to track pending conditional txs that would invalidate each other intrablock which would be pretty complex to implement